app/assets/javascripts/actions/campaign_actions.js

```diff
@@ -84,9 +84,38 @@ export const addCampaign = (courseId, campaignId) => dispatch => {
     addCampaignsPromise(courseId, campaignId)
       .then(data => {
         dispatch({
-          type: ADD_USER,
+          type: ADD_CAMPAIGN,
           data
         });
       })
   );
 };
+
+const fetchAllCampaignsPromise = () => {
+  return new Promise((res, rej) => {
+    return $.ajax({
+      type: 'GET',
+      url: `/all_campaigns`,
```

Comment:
I would like to know your opinion on this, @ragesoss 

I don't know if I am overcomplicating this issue, but I think we need to also retrieve all the campaigns available so we can add new more. What do you think to add an action that fetches them? Or would you do it differently?